geps/gep-1897.md

```diff
@@ -125,7 +125,231 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+    // status delivered in the response for validation failures.
+    Port PortNumber `json:port,omitempty`
+}
+
+// TLSConnectionPolicyStatus defines the observed state of TLSConnectionPolicy.
+type TLSConnectionPolicyStatus struct {
+    // Conditions describe the current conditions of the TLSConnectionPolicy.
+    //
+    // Implementations should prefer to express TLSConnectionPolicy
+    // conditions using the `TLSConnectionPolicyConditionType` and
+    // `TLSConnectionPolicyConditionReason` constants so that 
+    // operators and tools can converge on a common vocabulary to 
+    // describe TLSConnectionPolicy state.
+    // Known condition types are:
+    // 
+    // * “Accepted”
+    // 
+    // +optional
+    // +listType=map
+    // +listMapKey=type
+    // +kubebuilder:validation:MaxItems=8
+    // +kubebuilder:default={type: "Accepted", status: "Unknown", reason:"Pending", message:"Waiting for validation", lastTransitionTime: "1970-01-01T00:00:00Z"}
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// TLSConnectionPolicyConditionType is the type of a condition used
+// as a signal by TLSConnectionPolicy.  This type should be used with
+// the TLSConnectionPolicyStatus.Conditions field.
+type TLSConnectionPolicyConditionType string
+
+//  TLSConnectionPolicyConditionReason is a reason that explains why a
+// particular TLSConnectionPolicyConditionType was generated.
+type TLSConnectionPolicyConditionReason string
+
+const (
+    // This condition indicates that the TLSConnectionPolicy has been
+    // accepted as valid.
+    // Possible reason for this condition to be True is:
+    // * “Validated”
+    // Possible reasons for this condition to be False are:
+    // * “Invalid”
+    // * “Pending”
+    TLSConnectionPolicyConditionAccepted TLSConnectionPolicyConditionType = “Accepted”
+
+	// This reason is used with the “Accepted” condition when the condition is true.
+    TLSConnectionPolicyReasonAccepted TLSConnectionPolicyConditionReason = “Valid”
+
+    // This reason is used with the “Accepted” condition when the TLSConnectionPolicy is invalid, e.g. crossing namespace boundaries.
+    TLSConnectionPolicyReasonInvalid TLSConnectionPolicyConditionReason = “Invalid”
+
+    // This reason is used with the “Accepted” condition when the TLSConnectionPolicy is pending validation.
+    TLSConnectionPolicyReasonPending TLSConnectionPolicyConditionReason = “Pending”
+)
+```
+
+## How a client behaves
+
+This table describes the effect that a TLSConnectionPolicy has on a Route.  There are only two cases where the
+TLSConnectionPolicy will signal a Route to connect to a backend using TLS, an HTTPRoute with a backend that is targeted
+by a TLSConnectionPolicy, either with or without listener TLS configured.  (There are a few other cases where it may be
+possible, but is purposely marked “not supported” due to a desire for less confusion on the assigned purpose of each of
```

Comment:
What does an implementation do for "not supported"?

Reply:
In my understanding, these are invalid configurations, so they wouldn't pass validation.

Reply:
how can they fail validation? The configuration and the route are separate objects

Reply:
That is true, I didn't think it through.  So for "not supported", the intention really is that this is not supported, at least not in this proposal, where we're only covering HTTPRoute.

Reply:
This seems like something that would benefit from both nested conditions (as suggested in https://github.com/kubernetes-sigs/gateway-api/pull/2113/files#r1239031972) and the SupportedFeatures GEP from @LiorLieberman. We can say that every implementation that claims supports for this policy needs to populate status on the policy describing which Routes it is being implemented for (and _maybe_ also which Routes it is not being implemented for).

Reply:
How about, 

> Every implementation that claims supports for TLSConnectionPolicy should document for which Routes it is being implemented.